webapp/src/js/bootstrapper/index.js

```diff
@@ -224,53 +224,70 @@
       .catch(() => true);
 
     let isInitialReplicationNeeded;
-    Promise.all([swRegistration, testReplicationNeeded(), setReplicationId(POUCHDB_OPTIONS, localDb)])
-      .then(function(resolved) {
+    Promise
+      .all([swRegistration, testReplicationNeeded(), setReplicationId(POUCHDB_OPTIONS, localDb)])
+      .then(resolved => {
         purger.setOptions(POUCHDB_OPTIONS);
         isInitialReplicationNeeded = !!resolved[1];
 
         if (isInitialReplicationNeeded) {
+          window.startupTimes.replicationStarted = performance.now();
           return docCountPoll(localDb)
             .then(() => initialReplication(localDb, remoteDb))
             .then(testReplicationNeeded)
             .then(isReplicationStillNeeded => {
               if (isReplicationStillNeeded) {
                 throw new Error('Initial replication failed');
               }
-            });
+            })
+            .then(() => window.startupTimes.replicationEnded = performance.now());
         }
       })
       .then(() => {
         return purger
           .shouldPurge(localDb, userCtx)
           .then(shouldPurge => {
+            window.startupTimes.purgedDB = shouldPurge;
+
             if (!shouldPurge) {
               return;
             }
 
+            window.startupTimes.purgeStarted = performance.now();
```

Comment:
Do we have to save the start time in window.startupTimes? Can we just save it as a local variable and only save the duration in window.startupTimes?